### PR TITLE
types: change type of `ApplicationCommandSubCommand`.options

### PR DESCRIPTION
**Pull request by ZyXProFR**
**Please describe the changes this PR makes and why it should be merged:**

This PR changes the type of the `ApplicationCommandSubCommand.options` field to `ApplicationCommandOption[]` in order to match the documentation and be able to use all the properties of this type (like `minValue`, `maxValue`, etc).

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes

### CODE REVIEW

**Review by Jiralite — CHANGES_REQUESTED** [review]
This will introduce the issue where the type of the options will allow a subcommand group or a subcommand as a valid type. The current code does not allow this to happen as it is strongly typed as to what can be allowed. This needs to be modified - perhaps #8443 will help with this.

**Comment by ZyXProFR** [comment]
> This will introduce the issue where the type of the options will allow a subcommand group or a subcommand as a valid type. The current code does not allow this to happen as it is strongly typed as to what can be allowed. This needs to be modified - perhaps #8443 will help with this.

Hmm can you confirm that what you meant was my last commit?

**Comment by Jiralite** [comment]
It's better, but it's missing other option types now (which was always the case). #8443 partially fixes this, so either you can suggest the change there or you will have to wait until that pull request is merged, fetch the changes and add the missing types.

**Comment by ZyXProFR** [comment]
> It's better, but it's missing other option types now (which was always the case). #8443 partially fixes this, so either you can suggest the change there or you will have to wait until that pull request is merged, fetch the changes and add the missing types.

Okay thanks :)

**Comment by ZyXProFR** [comment]
Waiting for the merge of another pull request

**Pull request closed without merging**
